fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/services/BatchService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.services;
+
+import org.fcrepo.kernel.api.FedoraSession;
+
+/**
+ * @author acoburn
+ * @since Sept 30, 2016
+ */
+public interface BatchService {
+
+    /**
+     * Check for expired batch operations and remove them
+     */
+    void removeExpired();
+
+    /**
+     * Create a new batch operation with a FedoraSession and add it to the currently open ones
+     *
+     * @param session The session to use for this batch operation
+     * @param username the name of the {@link java.security.Principal}
+     */
+    void begin(FedoraSession session, String username);
```

Comment:
You are correct: the jcr.Session interface has a getUserID method, and the new FedoraSession interface has a getUserId method -- which in the impl just proxies to the jcr.Session::getUserID. The issue is that in the case of an anonymous user, the result of the jcr.Session::getUserID is not `null`, but rather "<anonymous>". But we can't actually rely on that string, because that, too, is configurable in the repository.json file. This way, we are relying on the value of the java.security.Principal::getName value for looking up FedoraSession values.

Believe me -- my first stab at this excluded the `username` parameter on all of these methods, but relying on that conflicts with how transaction values are looked up.